pkg/scheduler/framework/plugins/interpodaffinity/plugin.go

```diff
@@ -221,11 +221,35 @@ func (pl *InterPodAffinity) isSchedulableAfterNodeChange(logger klog.Logger, pod
 		return framework.Queue, err
 	}
 
+	// When queuing this Pod:
+	// - 1. The original node does not have the affinity key but the modified node does, the pod may become schedulable.
+	// - 2. Both the original and modified nodes have the affinity key and they differ, the pod may become schedulable.
+	// - 3. A new node is added with the affinity key, the pod may become schedulable.
 	for _, term := range terms {
```

Comment:
nit: better to put if originalNode == nil first to remove `else`

```go
for _, term := range terms {

if originalNode == nil {
	if _, ok := modifiedNode.Labels[term.TopologyKey]; ok {
		// Case 3: A new node is added with the affinity key.
		logger.V(5).Info("A node with a matched pod affinity topologyKey was added and it may make the pod schedulable",
			"pod", klog.KObj(pod), "node", klog.KObj(modifiedNode))
		return framework.Queue, err
	}

	continue
}

// .. handle the update event.
```

Reply:
done

Reply:
You can consider putting a !modifiedHasKey first (potentially consider node deletion as well), as this is the most clear eliminator. Currently every condition has modifiedHasKey check baked in somehow

if modifiedNode == nil || !modifiedHasKey { // node deleted or lablel disappeared
  continue
}
if originalNode == nil || !originalHasKey || originalTopologyKey != modifiedTopologyKey
  return Queue
}

For anti-affinity we probably need opposite:

if originalNode == nil || !originalHasKey {
  continue
}
if modifiedNode == nil || !modifiedHasKey || originalTopologyKey != modifiedTopologyKey
  return Queue
}

Obviously, since probably want to read topologyKey label once, we need to either break this flow into 5 different if statements or reading its value upfront